### ISSUE TITLE
[data grid] filter popup causes horizontal scrolling and extends further from website limits on mobile

### ISSUE THREAD

**Issue by maio4** [mui/mui-x#6143]
### Current behavior 😯

When clicking on the filter button in the DataGrid , the popup doesn't adjust to the screen limits and causes horizontal scrolling.
This image represents this page: https://v4.mui.com/components/data-grid/filtering/
![image](https://user-images.githubusercontent.com/113436409/189887653-cbf367fd-a456-46ce-a05e-41ba49c5375d.png)

### Expected behavior 🤔

I guess the expected behavior is some sort of wrapping of this component.

### Steps to reproduce 🕹

Steps:

1. Click on the filter button in the DataGrid on mobile

**Comment by alexfauquette** [comment]
Yes, the filter panel has a min-width. You can customize the [filter panel](https://mui.com/x/react-data-grid/filtering/#custom-filter-panel) CSS to override this behavior

Somewhat liked to the RFC #3719 About having a new design for filter panel

**Comment by michelengelen** [comment]
Closing this as part of a cleanup of stale issues targeting unsupported versions — this report is against the v4-era docs, and that line is end-of-life. The behavior itself is unchanged in nature: the filter panel inputs have a min-width, and on narrow viewports the answer from this thread still applies — override it via CSS or `slotProps.filterPanel.filterFormProps` to let the form shrink or wrap.

A filter panel that adapts to small screens out of the box is part of the panel redesign tracked in #6419 (the successor of the #3719 RFC linked above) — please add your 👍 and mobile use case there. If the panel still overflows the viewport on a supported version (v8/v9), a new issue with a reproduction is welcome.
